packages/flutter/lib/src/widgets/router.dart

```diff
@@ -512,23 +510,29 @@ class _RouterState<T> extends State<Router<T>> with RestorationMixin {
     _routeInformationReportingTaskScheduled = false;
 
     if (_routeInformation.value != null) {
-      final RouteInformation routeInformation = _routeInformation.value!;
+      final RouteInformation oldRouteInformation = widget.routeInformationProvider!.value;
+      final RouteInformation currentRouteInformation = _routeInformation.value!;
       switch (_currentIntentionToReport) {
         case _IntentionToReportRouteInformation.none:
           assert(false, '_reportRouteInformation must not be called with _IntentionToReportRouteInformation.none');
           return;
         case _IntentionToReportRouteInformation.ignore:
+          if (oldRouteInformation.location != currentRouteInformation.location ||
+              oldRouteInformation.state != currentRouteInformation.state) {
+            widget.routeInformationProvider!.routerUpdatesRouteInformation(currentRouteInformation);
```

Comment:
`Router.neglect` (which triggers this case) is documented to not report anything back to the engine, but now it does?

I think the previous intention was that you use `Router.neglect` if you don't want the URL in the browser to change? But now it would change, right?

What behavior do we want here? We should document it clearly.

Reply:
After looking at how complex the browser history management, i intend to make the neglect to update the URL but not push the new history entry. keeping the routeinformation in routeinformationprovider and browserhistory out of sync may give us more issue later.